### PR TITLE
XRE-2696 Change TrackedPoseDriver action property types to InputActionProperty

### PR DESCRIPTION
**Pull request by chris-massie**
Replaced the Position and Rotation Actions of `TrackedPoseDriver` with new properties of type `InputActionProperty` to allow references to actions to be used instead of directly serializing the `InputAction`. The previous values will be automatically migrated to the new format. Callers will get obsolete warnings when calling the old properties.

This PR also fixes undesired behavior where the property setters of the action properties would enable the actions even outside play mode in the Unity editor. It also now handles pushing the pose to the Transform when the tracked device reports `Vector3.zero` and `Quaternion.identity`.

Also added xml-docs and fixed some typos elsewhere.

Performed manual tests by opening a project that contained the Tracked Pose Driver with position and rotation actions set and verified that the behavior automatically migrated the data to the new format. Also tested setting the actions using the deprecated property setters and verified that the behavior serializes in the new format.

### CODE REVIEW

**Review by mattdatwork — APPROVED** [review]

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1234) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1234) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/InputSystem?pullRequest=1234) it.</sub>

**Comment by ekcoh** [comment]
Thank you for this PR @chris-massie. Could you please merge latest develop into this branch and possibly merge https://github.com/Unity-Technologies/InputSystem/pull/1320 into this branch (based off this branch). Feel free to ping me if you need support from our side. 

Ping me when conflicts have been resolved and I will help approving/merging this so we can close this long lived branch.

FYI @mattdatwork

**Comment by ekcoh** [comment]
@bryandube @chris-massie Seems to be an outstanding issue with 1 failing test, log line 2702 that needs some investigation:

```
[17:30:08.582 Information] [56/1360] TestFinish
	name: APIVerificationTests.API_MinorVersionsHaveNoBreakingChanges
	result: FAILED
	message:   Expected: <empty>
  But was:  < "    public class TrackedPoseDriver : UnityEngine.MonoBehaviour", "        public UnityEngine.InputSyste
[17:30:08.589 Information] m.InputAction positionAction { get; set; }", "        public UnityEngine.InputSystem.InputAction rotationAction { get; set; }" >

	stackTrace: at APIVerificationTests.API_MinorVersionsHaveNoBreakingChanges () [0x001f7] in C:\build\output\Unity-Technologies\InputSystem\Assets\Tests\InputSystem\APIVerificationTests.cs:818

	duration: 2.553465 seconds
```

**Comment by bryandube** [comment]
> @bryandube @chris-massie Seems to be an outstanding issue with 1 failing test, log line 2702 that needs some investigation:
> 
> ```
> [17:30:08.582 Information] [56/1360] TestFinish
> 	name: APIVerificationTests.API_MinorVersionsHaveNoBreakingChanges
> 	result: FAILED
> 	message:   Expected: <empty>
>   But was:  < "    public class TrackedPoseDriver : UnityEngine.MonoBehaviour", "        public UnityEngine.InputSyste
> [17:30:08.589 Information] m.InputAction positionAction { get; set; }", "        public UnityEngine.InputSystem.InputAction rotationAction { get; set; }" >
> 
> 	stackTrace: at APIVerificationTests.API_MinorVersionsHaveNoBreakingChanges () [0x001f7] in C:\build\output\Unity-Technologies\InputSystem\Assets\Tests\InputSystem\APIVerificationTests.cs:818
> 
> 	duration: 2.553465 seconds
> ```

Looks like the tracked pose driver change is tripping the semver for some reason.

**Comment by bryandube** [comment]
> @bryandube @chris-massie Seems to be an outstanding issue with 1 failing test, log line 2702 that needs some investigation:
> 
> ```
> [17:30:08.582 Information] [56/1360] TestFinish
> 	name: APIVerificationTests.API_MinorVersionsHaveNoBreakingChanges
> 	result: FAILED
> 	message:   Expected: <empty>
>   But was:  < "    public class TrackedPoseDriver : UnityEngine.MonoBehaviour", "        public UnityEngine.InputSyste
> [17:30:08.589 Information] m.InputAction positionAction { get; set; }", "        public UnityEngine.InputSystem.InputAction rotationAction { get; set; }" >
> 
> 	stackTrace: at APIVerificationTests.API_MinorVersionsHaveNoBreakingChanges () [0x001f7] in C:\build\output\Unity-Technologies\InputSystem\Assets\Tests\InputSystem\APIVerificationTests.cs:818
> 
> 	duration: 2.553465 seconds
> ```

Looks like a property was removed.  @chris-massie  was the original author of this one so we will probably need to rework this in some way to prevent the need for a major bump.

**Comment by ekcoh** [comment]
@bryandube I can consult the Input team as well tomorrow morning for guidance. Not familiar with these tests/checks yet.

**Comment by bryandube** [comment]
> @bryandube I can consult the Input team as well tomorrow morning for guidance. Not familiar with these tests/checks yet.

That check is basically looking for breaking changes, something that if the user were to update to this their code may not compile, etc.  This change has removed a property which means anyone who was referencing it would now be broken.  This is only ok in a package if you update the major version number, aka, going to 2.0.0.  Since this is likely not an option we will need to rework these changes to fix it.  Chris is out this week but I will take a look into whether it can be fixed.

**Comment by bryandube** [comment]
> @bryandube I can consult the Input team as well tomorrow morning for guidance. Not familiar with these tests/checks yet.

I may have been wrong about those properties being gone too, I think they are still there in a partial class, so looking closer into why this is failing.

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ however, it has to be formatted properly to pass verification tests.
 
 ### Changed
 
+- Changed `TrackedPoseDriver` to use properties of type `InputActionProperty` rather than `InputAction` to allow more
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Looks like the end of the sentence is missing?

Comment (bryandube) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Fixed, also found there was a merge error that wasnt building and fixed it.  That got me thinking that I am not sure whether this change has had an coverage from the quality team so I added @jackpr-unity to the PR.

**Review by jackpr-unity — CHANGES_REQUESTED** [review]
Can you add a test that checks the data route for position and rotation? That is: set the action references, change the source binding's value, and check that the corresponding transform actually gets these new values?

**Comment by bryandube** [comment]
Tests are now all green except for the CLA.

**Comment by JoshPerson** [comment]
Just tested input with Quest 2 (linked) on OpenXR.  All tracking and input works correctly as far as I can tell.

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ however, it has to be formatted properly to pass verification tests.
 
 ### Changed
 
+- Changed `TrackedPoseDriver` to use properties of type `InputActionProperty` rather than `InputAction` to allow more flexibility.
+- XRLayoutBuilder now supports `_` in sanitized names.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/CHANGELOG.md:16]: Sidenote: neither of these is particularly helpful/useful.

The first entry would ideally highlight how this increased flexibility and what in particular was impacted. The second entry mentions an internal API that is meaningless to the user. Likely, the `_ in sanitized names` part probably doesn't mean anything to the user.

**Pull request merged**